### PR TITLE
Test runner for accessibility testing

### PR DESCRIPTION
**Pull request by ana-buildt**
Add [Pa11y-CI](https://github.com/pa11y/pa11y-c) accessibility test runner.

### CODE REVIEW

**Comment by ThePeach** [comment]
works and looks good... what are we going to do with all the errors? Is this going to be integrated into travis once merged or is there more work that needs to be done?

**Comment by ana-buildt** [comment]
I'll check the errors, it would probably worth fixing them before to be integrated into Travis.

File: templates/partials/page-footer.hbs [hunk 1]
```diff
@@ -1,22 +1,32 @@
 <footer role="contentinfo"{{#if footerModifierClass}} class="{{footerModifierClass}}"{{/if}}>
-  <p>made with ❤️ <a href="/"><svg role="img" class="grav-c-logo" aria-label="buildit @ wipro digital" width="300" height="33"><use xlink:href="#buildit-logotype"></use></svg></a></p>
+  <p>made with ❤️ 
+    <a href="/">
+      <title>Buildit Logo</title>
```
Comment (ThePeach) [inline templates/partials/page-footer.hbs]: shouldn't the title go inside the `<svg>` tag?

Comment (ThePeach) [reply templates/partials/page-footer.hbs]: We do have `<title>`s and `<description>` in our source SVGs so I'm guessing the compressor is removing them for some odd reason, which shouldn't be the case... @james-nash any ideas?

Comment (james-nash) [reply templates/partials/page-footer.hbs]: @ThePeach @ana-buildt 
So, `<title>` definitely needs to be within the `<svg>`.

We're using `gulp-svg-symbols` to generate our symbol defs. You're right, the source files do have correct `<title>`s in them. Not sure why those aren't being preserved. Let me investigate and get back to you.

Comment (james-nash) [reply templates/partials/page-footer.hbs]: So, the SVG symbol defs from Gravity _do_ include the `<title>`s (do a view source on the website page to see for yourself, they're all there).

The trouble seems to be that browser's will not expose the `<title>` (or `<description>`) of an SVG being referenced by `<use>`. Therefore, screen readers can't access that info either.

It seems that, wherever you reference an SVG symbol, you need to either include a `<title>` there directly. Or, you can reference the symbol's one (if it has an ID) via an `aria-labelledby` attribute.

See:
* https://css-tricks.com/accessible-svgs/
* https://stackoverflow.com/questions/38353094/making-svg-symbol-accessible

Currently, our SVG titles do not have ID attributes.

I will have a look at `gulp-svg-symbols` in Gravity, to see if that's something we can easily rectify there. Assuming it is, you can use the `aria-labelledby` trick.

Until then, your only option for the website will be to manually add `<title>` attributes. Given that they only occur infrequently, that might be the best for MVP.

Comment (james-nash) [reply templates/partials/page-footer.hbs]: So, to clarify, where you have something like:
```html
<svg role="img" class="..." width="..." height="...">
    <use xlink:href="#buildit-foo"></use>
</svg>
```

...it should be something like:
```html
<svg role="img" class="..." width="..." height="...">
    <title>A picture of a unicorn pooping rainbows</title>
    <use xlink:href="#buildit-foo"></use>
</svg>
```

File: package.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
     "test-watch": "jest --coverage --watch",
     "cypress:open": "cypress open",
     "cypress:cli": "cypress run",
-    "cypress:travis": "http-server dist -c-1 -s & cypress run"
+    "cypress:travis": "http-server dist -c-1 -s & cypress run",
+    "pa11y-run": "http-server dist -c-1 -s & ./node_modules/.bin/pa11y-ci"
```
Comment (james-nash) [inline package.json]: Please add a `prepa11y-run` script which runs `npm run-script build`, to ensure that the a11y tests are always run on a fresh build of the website.

See NPM's `package.json` docs for more info: https://docs.npmjs.com/misc/scripts

**Comment by ThePeach** [comment]
no errors! :tada: sweet!

File: package.json (hunk 2 above)
Comment (ThePeach) [reply package.json]: and update the docs to reflect this change too!

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
   "eslint.enable": true,
-  "eslint.autoFixOnSave": true,
+  "eslint.autoFixOnSave": false,
```
Comment (james-nash) [inline .vscode/settings.json]: Sorry, I didn't spot this before. Is this correct? I thought we were intentionally running Prettier on save to ensure JS code conforms with our conventions.

**Pull request closed without merging**